platform-sdk/swirlds-jasperdb/src/main/java/com/swirlds/merkledb/files/DataFileWriter.java

```diff
@@ -149,9 +135,35 @@ public DataFileWriter(
      * @throws IOException if I/O error(s) occurred
      */
     private void moveMmapBuffer(final int currentMmapPos) throws IOException {
-        mmapPositionInFile += currentMmapPos;
-        closeMmapBuffer();
-        writingMmap = writingChannel.map(MapMode.READ_WRITE, mmapPositionInFile, MMAP_BUF_SIZE);
+        try (final FileChannel channel = FileChannel.open(path, StandardOpenOption.READ, StandardOpenOption.WRITE)) {
+            final MappedByteBuffer newMap =
+                    channel.map(MapMode.READ_WRITE, mmapPositionInFile + currentMmapPos, MMAP_BUF_SIZE);
+            verifyMmapCorrectness(newMap);
+            closeMmapBuffer();
+            writingMmap = newMap;
+            mmapPositionInFile += currentMmapPos;
+        }
+    }
+
+    /**
+     * This method addresses the fact that the mapped byte buffer doesn't have the same safeguards
+     * as the regular FileChannel. It's possible to create a mapped byte buffer even if there is not enough
+     * disk space to map it. This code is trying to identify this situation and throw an exception.
+     *
+     * @param newMap the mapped byte buffer to verify
+     */
+    private static void verifyMmapCorrectness(final MappedByteBuffer newMap) throws IOException {
+        // theoretically it's possible, we should check it
+        if(newMap == null) {
+            throw new IOException("Failed to map file channel to memory");
+        }
+        // The idea is to force this mapping to happen. By default, actual mapping happens lazily.
+        try {
+            newMap.put(MMAP_BUF_SIZE - 1, (byte) -1);
+            if (newMap.get(MMAP_BUF_SIZE - 1) != -1) throw new IOException("Fatal error when creating mmap. Possibly,  out of disk memory.");
```

Comment:
Does spotless allow single-line "if" statements? Even if so, I'd put the "throw" statement into a { } block to make it easier to read.

Reply:
The line looks too long anyway. Also, double whitespace after "Possibly,"

Reply:
Fixed